src/routes/CreateNewSafePage/CreateNewSafePage.tsx

```diff
@@ -0,0 +1,195 @@
+import { sm } from 'src/theme/variables'
+import StepperForm, { StepFormElement } from 'src/components/StepperForm/StepperForm'
+import NameNewSafeStep, { nameNewSafeStepLabel } from './steps/NameNewSafeStep'
+import { APP_ENV } from 'src/utils/constants'
+import {
+  CreateSafeFormValues,
+  FIELD_CREATE_CUSTOM_SAFE_NAME,
+  FIELD_CREATE_SUGGESTED_SAFE_NAME,
+  FIELD_MAX_OWNER_NUMBER,
+  FIELD_NEW_SAFE_PROXY_SALT,
+  FIELD_NEW_SAFE_THRESHOLD,
+  FIELD_SAFE_OWNERS_LIST,
+  SAFE_PENDING_CREATION_STORAGE_KEY,
+} from './fields/createSafeFields'
+import { useMnemonicSafeName } from 'src/logic/hooks/useMnemonicName'
+import { providerNameSelector, userAccountSelector } from 'src/logic/wallets/store/selectors'
+import OwnersAndConfirmationsNewSafeStep, {
+  ownersAndConfirmationsNewSafeStepLabel,
+  ownersAndConfirmationsNewSafeStepValidations,
+} from './steps/OwnersAndConfirmationsNewSafeStep'
+import { currentNetworkAddressBookAsMap } from 'src/logic/addressBook/store/selectors'
+import ReviewNewSafeStep, { reviewNewSafeStepLabel } from './steps/ReviewNewSafeStep'
+import { useLocation } from 'react-router'
+import { loadFromStorage, saveToStorage } from 'src/utils/storage'
+import SafeCreationProcess from './components/SafeCreationProcess'
+import { Loader } from '@gnosis.pm/safe-react-components'
+import SelectWalletAndNetworkStep, { selectWalletAndNetworkStepLabel } from './steps/SelectWalletAndNetworkStep'
+
+function CreateNewSafePage(): ReactElement {
+  const classes = useStyles()
+
+  const [safePendingToBeCreated, setSafePendingToBeCreated] = useState<CreateSafeFormValues | null>()
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    async function checkIfSafeIsPendingToBeCreated() {
+      setIsLoading(true)
+      const safePendingToBeCreated = (await loadFromStorage(SAFE_PENDING_CREATION_STORAGE_KEY)) as CreateSafeFormValues
+      setSafePendingToBeCreated(safePendingToBeCreated)
+      setIsLoading(false)
+    }
+    checkIfSafeIsPendingToBeCreated()
+  }, [])
+
+  const provider = useSelector(providerNameSelector)
+  const userWalletAddress = useSelector(userAccountSelector)
+  const addressBook = useSelector(currentNetworkAddressBookAsMap)
+  const location = useLocation()
+  const safeRandomName = useMnemonicSafeName()
+
+  async function showSafeCreationProcess(newSafeFormValues: CreateSafeFormValues) {
+    await saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, { ...newSafeFormValues })
+    setSafePendingToBeCreated(newSafeFormValues)
+  }
+
+  const [initialFormValues, setInitialFormValues] = useState<CreateSafeFormValues>()
+
+  useEffect(() => {
+    if (provider && userWalletAddress) {
+      const initialValuesFromUrl = getInitialValues(userWalletAddress, addressBook, location, safeRandomName)
+      setInitialFormValues(initialValuesFromUrl)
+    }
+  }, [provider, userWalletAddress, addressBook, location, safeRandomName])
+
+  const isProductionEnv = APP_ENV === 'production'
```

Comment:
Don't we already have something like this exported somewhere?

Reply:
mmm, where? I checked other components and we are using something like `const isProductionEnv = APP_ENV === 'production'`

Reply:
`src/utils/constants.ts` => `IS_PRODUCTION`

Reply:
`IS_PRODUCTION` is checking that `export const IS_PRODUCTION = NODE_ENV === 'production'` we need to check that ```process.env.REACT_APP_ENV === 'production'```

Reply:
Maye move this to the constants file?